### PR TITLE
Feat: Implement Command line functionality & featureset 

### PR DESCRIPTION
**Pull request by ingrettable**
began implementation of clap CLI utility. changed terminal UI by the following commands: producer register , producer add <FILE_NAME>, consumer upload <FILE_NAME> and consumer get .

Authors: @daminals @ingrettable

TODOs:
- communicate with Desktop team in order to design the API
- finish testing

### CODE REVIEW

**Review by daminals — APPROVED** [review]
After doing some minor testing it looks like everything is working great and ready to integrate with the desktop team

**Comment by xiej2520** [comment]
Tested each CLI feature and it all looks like it works, no major concerns about code quality, LGTM.

**Review by andrewctam — APPROVED** [review]
LGTM

File: peernode/src/peer-node/cli/tests.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    #[test]
+    fn test_clap_cli() {
+        // Simulate command line arguments
+        let args = vec!["market", "set", "localhost:50051"];
+
+        // Call your CLI function with the arguments
+        let result = cli().get_matches_from(args);
+
+        // Assert the expected output
+        //assert_eq!(result, Some(_));
```
Comment (devu1997) [inline peernode/src/peer-node/cli/tests.rs]: Is this intended? The test will not be complete without assertion right? If you are planning on fixing this later then add a TODO comment.

File: peernode/src/peer-node/client.rs [hunk 2]
```diff
@@ -1,49 +1,79 @@
...
+    if std::env::args().len() > 1 {
+        // remove the first argument which is the name of the program
+        let args = std::env::args().skip(1).collect::<Vec<String>>();
+        let matches = cli.clone().get_matches_from(args);
+        match handle_arg_matches(matches, &mut config).await {
+            Ok(_) => {}
+            Err(e) => eprintln!("\x1b[93mError:\x1b[0m {}", e),
+        };
+        // wait for the HTTP server to start
+        // tokio::time::sleep(tokio::time::Duration::from_secs(5)).await;
```
Comment (devu1997) [inline peernode/src/peer-node/client.rs]: Remove this if not required. Or add TODO with explanation on why this is kept as a comment

File: peernode/src/peer-node/producer/mod.rs [hunk 3]
```diff
@@ -44,28 +62,25 @@ pub async fn run(market: String, ip: Option<String>, port: Option<u16>) -> Resul
...
-    // Register the files with the market service
-    let hash = file_map.get_hashes().await;
-    for hash in hash {
-        // TODO: Find a way to get the public IP address
-        println!("Producer: Registering file with hash {}", hash);
+    for (hash, price) in prices {
+        println!(
+            "Producer: Registering file with hash {} and price {}",
+            hash, price
+        );
```
Comment (devu1997) [inline peernode/src/peer-node/producer/mod.rs:72]: The println is short enough to fit in one line right?

File: peernode/src/peer-node/cli/tests.rs (hunk 1 above)
Comment (daminals) [reply peernode/src/peer-node/cli/tests.rs]: Got it, will do

File: peernode/src/peer-node/producer/mod.rs (hunk 3 above)
Comment (daminals) [reply peernode/src/peer-node/producer/mod.rs:72]: Screenshot 2024-04-15 at 6 43 16 PM

Comment (devu1997) [reply peernode/src/peer-node/producer/mod.rs:72]: I meant to fit in one line in the code.
`println!("Producer: Registering file with hash {} and price {}", hash, price);`

Comment (daminals) [reply peernode/src/peer-node/producer/mod.rs:72]: This is using the auto formatter built into cargo "cargo fmt"

**Pull request merged**